src/Engine/EmitMSIL/CodeGenIL.cs

```diff
@@ -611,8 +623,14 @@ private Type EmitFunctionCallNode(AssociativeNode node)
             methodName = fcn.Function.Name;
             var args = fcn.FormalArguments;
             var numArgs = args.Count;
+            isInternal = CoreUtils.IsInternalMethod(methodName);
 
-            if(compilePass == CompilePass.MethodLookup)
+            if (isInternal)
+            {
+                className = nameof(Internals);
+            }
+
```

Comment:
Why don't we call this inside FunctionLookup itself? This way we won't need to store `isInternal` as a class member.

Reply:
The reason is that `KeyGen` needs a consistent class name and I tried to avoid calling `CoreUtils.IsInternalMethod(methodName)` inside that method but that is maybe OK?